### PR TITLE
fix(core): throw if standalone components are present in `@NgModule.bootstrap`

### PR DESCRIPTION
**Pull request by AndrewKushnir**
This commit updates the logic to detect a situation when a standalone component is used in the NgModule-based bootstrap (`@NgModule.bootstrap`). Both AOT and JIT compilers are updated to handle this situation.

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by pkozlowski-opensource — DISMISSED** [review]
LGTM

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:445656923:BASE:445656934:1651344286889:ffe7e7fc).

File: goldens/public-api/compiler-cli/error_code.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ export enum ErrorCode {
     INVALID_BANANA_IN_BOX = 8101,
     MISSING_PIPE = 8004,
     MISSING_REFERENCE_TARGET = 8003,
+    NGMODULE_BOOTSTRAP_IS_STANDALONE = 6008,
     NGMODULE_DECLARATION_IS_STANDALONE = 6008,
```
Comment (alfaproject) [inline goldens/public-api/compiler-cli/error_code.md:51]: 6008 is repeated. Maybe worth configuring this rule? https://typescript-eslint.io/rules/no-duplicate-enum-values/

Comment (AndrewKushnir) [reply goldens/public-api/compiler-cli/error_code.md:51]: Good catch, thanks! I've changed the error code and will look into configuring the lint rule.

**Review by dylhunn — DISMISSED** [review]
reviewed-for: public-api

File: packages/core/src/render3/jit/module.ts [hunk 2]
```diff
@@ -338,6 +338,14 @@ function verifySemanticsOfNgModuleDef(
     if (!getComponentDef(type)) {
       errors.push(`${stringifyForError(type)} cannot be used as an entry component.`);
     }
+    if (isStandalone(type)) {
```
Comment (atscott) [inline packages/core/src/render3/jit/module.ts:341]: Are these `jit` checks protected by an NG_DEV_MODE condition or are all of these verifications present in the prod bundle?

**Review by atscott — APPROVED** [review]
reviewed-for: public-api

File: packages/core/src/render3/jit/module.ts (hunk 2 above)
Comment (AndrewKushnir) [reply packages/core/src/render3/jit/module.ts:341]: It's a part of the JIT compiler, so it wouldn't make it into the bundle unless JIT is used (and we don't specifically optimize bundle size for JIT at this moment).

**Comment by AndrewKushnir** [comment]
Merge-assistance: the scope of the `public-api` approval is only a new error code, so 2 approvals should be sufficient.

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit fde4942cdf5133119b13ed26ee2f6976b787d84c.

**Pull request closed without merging**